presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,457 @@
+            return WriteMapping.doubleMapping("Float64", doubleWriteFunction());
+        }
+        if (isVarcharType(type)) {
+            return WriteMapping.sliceMapping("String", varcharWriteFunction());
+        }
+        if (type.getBaseName().equals("char")) {
+            return WriteMapping.sliceMapping("String", varcharWriteFunction());
+        }
+        return super.toWriteMapping(session, type);
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        try {
+            return createTable(session, tableMetadata, generateTemporaryTableName());
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
+
+    @Override
+    public JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, String tableName)
+            throws SQLException
+    {
+        SchemaTableName schemaTableName = tableMetadata.getTable();
+
+        JdbcIdentity identity = JdbcIdentity.from(session);
+        if (!getSchemaNames(identity).contains(schemaTableName.getSchemaName())) {
+            throw new PrestoException(NOT_FOUND, "Schema not found: " + schemaTableName.getSchemaName());
+        }
+
+        try (Connection connection = connectionFactory.openConnection(JdbcIdentity.from(session))) {
+            boolean uppercase = connection.getMetaData().storesUpperCaseIdentifiers();
+            String remoteSchema = toRemoteSchemaName(identity, connection, schemaTableName.getSchemaName());
+            String remoteTable = toRemoteTableName(identity, connection, remoteSchema, schemaTableName.getTableName());
+
+            if (uppercase) {
+                tableName = tableName.toUpperCase(ENGLISH);
+            }
+            String catalog = connection.getCatalog();
+
+            Map<String, Object> tableProperties = tableMetadata.getProperties();
+
+            ImmutableList.Builder<String> columnNames = ImmutableList.builder();
+            ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
+            ImmutableList.Builder<String> columnList = ImmutableList.builder();
+
+            for (ColumnMetadata column : tableMetadata.getColumns()) {
+                String columnName = column.getName();
+                if (uppercase) {
+                    columnName = columnName.toUpperCase(ENGLISH);
+                }
+                columnNames.add(columnName);
+                columnTypes.add(column.getType());
+                String typeStatement = toWriteMapping(session, column.getType()).getDataType();
+                if (column.isNullable()) {
+                    typeStatement = "Nullable(" + typeStatement + ")";
+                }
+                columnList.add(format("%s %s", columnName, typeStatement));
+            }
+
+            ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
```

Comment:
Can we modify `createTableSql` to pass table properties and generate the SQL from it ? We can add those changes as a separate commit